src/server/api/consume-wifi-data.ts

```diff
@@ -0,0 +1,109 @@
+import { Kafka } from "kafkajs";
+import { SchemaRegistry } from "@kafkajs/confluent-schema-registry";
+import { serverSupabaseClient } from "#supabase/server";
+import type { SupabaseClient } from '@supabase/supabase-js';
+
+const { internalSecret, kafkaConfig, schemaRegistry } = useRuntimeConfig();
+
+console.time("Initialize");
+const kafka = new Kafka({
+  clientId: "tudelft-spacefinder",
+  ssl: true,
+  sasl: {
+    mechanism: "plain",
+    username: kafkaConfig.username,
+    password: kafkaConfig.password,
+  },
+  brokers: [kafkaConfig.url],
+});
+
+const registry = new SchemaRegistry({
+  host: schemaRegistry.url,
+  auth: {
+    username: schemaRegistry.username,
+    password: schemaRegistry.password,
+  },
+});
+
+const consumer = kafka.consumer({ groupId: kafkaConfig.groupId });
+console.timeEnd("Initialize");
+
+let seeked = false;
+
+export default defineEventHandler(async (event) => {
+  if (getQuery(event)?.secret !== internalSecret) {
+    event.node.res.statusCode = 401;
+    event.node.res.end();
+    return;
+  }
+
+  const client = serverSupabaseClient(event);
+
+  consumeLastBatch({ client });
+
+  // Close consumer connection within function time limit
+  await new Promise((resolve) => setTimeout(async () => {
+    console.time("Consumer disconnect");
+    await consumer.disconnect();
+    console.timeEnd("Consumer disconnect");
+    event.node.res.statusCode = 202;
+    event.node.res.end();
+    resolve(null);
+  }, 20000));
+});
+
+async function consumeLastBatch({ client }: { client: SupabaseClient }) {
+  console.time("Consumer setup");
```

Comment:
Not really but having function input as an object is nice imo. It's more descriptive & flexible than ordered parameters.

Reply:
I've started to type stuff about "oh but that's a single parameter and while I agree in all other cases", but I guess it's OK here as well, just because the function name doesn't help us deduct what its parameter is.